pkg/server/router/manifest.go

```diff
@@ -313,20 +314,14 @@ func (mr ManifestRouter) SubmitApplication(ctx context.Context, w http.ResponseW
 		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusBadRequest)
 	}
 
-	submitApplicationResponse, err := mr.service.ProcessApplicationSubmission(*req)
+	op, err := mr.service.ProcessApplicationSubmission(*req)
 	if err != nil {
 		errMsg := "could not submit application"
 		logrus.WithError(err).Error(errMsg)
 		return framework.NewRequestError(errors.Wrap(err, errMsg), http.StatusInternalServerError)
 	}
 
-	resp := SubmitApplicationResponse{
-		Response:    submitApplicationResponse.Response,
-		Credentials: submitApplicationResponse.Credentials,
-		ResponseJWT: submitApplicationResponse.ResponseJWT,
-	}
-
-	return framework.Respond(ctx, w, resp, http.StatusCreated)
+	return framework.Respond(ctx, w, routerModel(*op), http.StatusCreated)
```

Comment:
eventually we can create a routerModel everywhere

Reply:
For `Operation` objects, yes!